forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabRepository.java

```diff
@@ -343,31 +343,36 @@ public void writeFileContents(String filename, String content, Branch branch, St
             // Use POST to create a new file
             request.post("repository/files/" + encodedFileName)
                     .body(body)
+                    .onError(writeFilecontentsOnError(filename, content, branch))
                     .execute();
         } else {
             // USE PUT to update the file
             request.put("repository/files/" + encodedFileName)
                     .body(body)
-                    .onError(response -> {
-                        // When GitLab returns 400, it may have still performed the update, so
-                        // need to check the current file contents.
-                        if (response.statusCode() == 400) {
-                            log.info("Received status code 400 when writing file " + filename
-                                    + " in repo " + name() + ", checking if file was stored anyway");
-                            var currentContents = fileContents(filename, branch.name());
-                            if (currentContents.isPresent() && content.equals(currentContents.get())) {
-                                // Need to return something other than empty
-                                log.info("Writing file " + filename + " in repo " + name()
-                                        + " was found to be successful in spite of return code 400");
-                                return Optional.of(JSON.of());
-                            }
-                        }
-                        return Optional.empty();
-                    })
+                    .onError(writeFilecontentsOnError(filename, content, branch))
                     .execute();
         }
     }
 
+    private RestRequest.ErrorTransform writeFilecontentsOnError(String filename, String content, Branch branch) {
```

Comment:
writeFileContentsOnError?(with Capital 'C')

Reply:
Looking at this again, having a separate method is unnecessary. I was a bit too quick refactoring this yesterday. Just creating a local instance of the lambda seems better.